### PR TITLE
Move functional tests related 3rd party libs into functional level

### PR DESCRIPTION
**Pull request by OscarQQ**
related: https://github.com/AdoptOpenJDK/TKG/issues/63
Signed-off-by: OscarQQ <Yixin.Qian@ibm.com>

### CODE REVIEW

File: scripts/getDependencies.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# limitations under the License.
+-->
+
+<project name="getDependency" default="build" basedir=".">
+	<taskdef resource="net/sf/antcontrib/antlib.xml" />
+	<description>
+		Get Third Party Dependencies
+	</description>
+
+	<!-- set properties for this build -->
```
Comment (renfeiw) [inline scripts/getDependencies.xml:23]: set default LIB property to all

File: makefile [hunk 2]
```diff
@@ -36,10 +36,10 @@ endif
 #######################################
 # download all dependent jars
 #######################################
-getdependency:
-	perl scripts$(D)getDependencies.pl -path '$(TEST_ROOT)$(D)TKG$(D)lib' -task default
+# getdependency:
```
Comment (renfeiw) [inline makefile:39]: don't forget to remove the comment section above

File: scripts/getDependencies.pl [hunk 3]
```diff
@@ -26,9 +26,11 @@
 my $path;
 # define task
 my $task = "default";
+my $dependencyList;
```
Comment (renfeiw) [inline scripts/getDependencies.pl:29]: You probably want to initialize it to "". Otherwise the code might break when there's no dependencyList option.

Comment (llxia) [reply scripts/getDependencies.pl:29]: Could we default to all? Only change if `dependencyList` is set by users. In this way, all the getDependency jobs do not need to update. i.e., https://ci.eclipse.org/openj9/job/test.getDependency/

File: scripts/build_tools.xml [hunk 4]
```diff
@@ -27,7 +27,14 @@
 		<mkdir dir="${build}" />
 	</target>
 
-	<target name="compile" depends="init" description="Using java ${JDK_VERSION} to compile the source  ">
+	<target name="getDependency" depends="init" description="get third party dependency">
+		<subant target="build">
+			<property name="LIB" value="json_simple"/>
+			<fileset dir="./" includes="getDependencies.xml" />
+		</subant>
+	</target>
```
Comment (llxia) [inline scripts/build_tools.xml:35]: If we use `<import file="./getDependencies.xml"/>`, I think we can avoid defining the above target.

File: scripts/getDependencies.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	<property name="LIB" value="all" />
+
+	<target name="build">
+		<exec executable="perl">
+			<arg line="getDependencies.pl -path ${TEST_ROOT}/TKG/lib -task default -dependencyList ${LIB}"/>
+		</exec>
+	</target>
```
Comment (llxia) [inline scripts/getDependencies.xml:30]: - rename `build ` to something meaningful. For example, getDependentLibs

File: scripts/build_tools.xml [hunk 6]
```diff
@@ -27,7 +27,14 @@
 		<mkdir dir="${build}" />
 	</target>
 
-	<target name="compile" depends="init" description="Using java ${JDK_VERSION} to compile the source  ">
+	<target name="getDependency" depends="init" description="get third party dependency">
+		<subant target="build">
+			<property name="LIB" value="json_simple"/>
```
Comment (OscarQQ) [inline scripts/build_tools.xml:32]: how could we define LIB if we are using import?

**Review by renfeiw — APPROVED** [review]
LGTM

**Pull request merged**